activerecord/test/cases/locking_test.rb

```diff
@@ -790,29 +790,34 @@ def test_no_locks_no_wait
     end
 
     private
-      def duel(zzz = 5, &block)
+      def duel(&block)
         t0, t1, t2, t3 = nil, nil, nil, nil
 
+        a_wakeup = Thread::Queue.new
+        b_wakeup = Thread::Queue.new
+
         a = Thread.new do
           t0 = Time.now
           Person.transaction do
             yield
-            sleep zzz       # block thread 2 for zzz seconds
+            b_wakeup << true
+            a_wakeup.pop
           end
           t1 = Time.now
         end
 
         b = Thread.new do
-          sleep zzz / 2.0   # ensure thread 1 tx starts first
+          b_wakeup.pop
```

Comment:
Ah, nice. I wasn't familiar with Thread::Queue#pop
> If the queue is empty, the calling thread is suspended until data is pushed onto the queue.
https://rubyapi.org/3.2/o/thread/queue#method-i-pop

So this waits until Thread `a` pushes `true` into `b_wakeup`.